app/ante/evm/04_validate.go

```diff
@@ -39,6 +39,11 @@ func checkDisabledCreateCall(
 	permissions *evmtypes.AccessControl,
 ) error {
 	to := txData.GetTo()
+	data := txData.GetData()
+	// If its not a contract creation or contract call this check is irrelevant
```

Comment:
But what about the `fallback` and `receive` functions of a contract?
An EOA can trigger a contract call by just sending funds to it.

We could identify this by checking if the to address is a contract, e.g. checking if the address has code

Reply:
this an important point that should be taken into consideration

Reply:
I suggest for contracts checking the method ID from the call data (i.e the first 4 bytes) and comparing it with the `receive` and `fallback` bytes